sdk/identity/azure-identity/src/main/java/com/azure/identity/WorkloadIdentityCredential.java

```diff
@@ -89,6 +92,13 @@ public class WorkloadIdentityCredential implements TokenCredential {
         ClientAssertionCredential tempClientAssertionCredential = null;
         String tempClientId = null;
 
+        if (identityClientOptions.isKubernetesTokenProxyEnabled()) {
```

Comment:
we should throw when this condition is false and any of the other proxy config variables has a value because that suggests the developer probably intends to use the proxy and has botched the configuration